I'm currently having issues with GCC 14 while there are no problems with GCC 13.

```
make -j1 V=sc toolchain/gcc/final/compile

...

bin/bash ./libtool  --tag=CC   --mode=link /fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0-final/./gcc/xgcc -B/fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0-final/./gcc/ -B/fork.openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/aarch64-openwrt-linux-gnu/bin/ -B/fork.openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/aarch64-openwrt-linux-gnu/lib/ -isystem /fork.openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/aarch64-openwrt-linux-gnu/include -isystem /fork.openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/aarch64-openwrt-linux-gnu/sys-include    -ftls-model=initial-exec -Wall -Werror  -Wc,-pthread  -O2 -pipe -mcpu=cortex-a53 -fno-caller-saves -fno-plt -fhonour-copts -Wformat -Werror=format-security -g -Os   -Wl,-O1    -o libitm.la -version-info 1:0:0 -Wl,--version-script,/fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0/libitm/libitm.map  -rpath /fork.openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/aarch64-openwrt-linux-gnu/lib aatree.lo alloc.lo alloc_c.lo alloc_cpp.lo barrier.lo beginend.lo clone.lo eh_cpp.lo local.lo query.lo retry.lo rwlock.lo useraction.lo util.lo sjlj.lo tls.lo method-serial.lo method-gl.lo method-ml.lo   futex.lo  
libtool: link: /fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0-final/./gcc/xgcc -B/fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0-final/./gcc/ -B/fork.openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/aarch64-openwrt-linux-gnu/bin/ -B/fork.openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/aarch64-openwrt-linux-gnu/lib/ -isystem /fork.openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/aarch64-openwrt-linux-gnu/include -isystem /fork.openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/aarch64-openwrt-linux-gnu/sys-include    -shared  -fPIC -DPIC .libs/libitm.la.lnkscript   -pthread -mcpu=cortex-a53 -Wl,-O1 -Wl,--version-script -Wl,/fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0/libitm/libitm.map   -Wl,-soname -Wl,libitm.so.1 -o .libs/libitm.so.1.0.0
libtool: link: rm -f  .libs/libitm.la.lnkscript
libtool: link: (cd ".libs" && rm -f "libitm.so.1" && ln -s "libitm.so.1.0.0" "libitm.so.1")
libtool: link: (cd ".libs" && rm -f "libitm.so" && ln -s "libitm.so.1.0.0" "libitm.so")
libtool: link: 
libtool: link: /fork.openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/aarch64-openwrt-linux-gnu/bin/ar --plugin /fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0-final/./gcc/liblto_plugin.so rc .libs/libitm.a aatree.o alloc.o alloc_c.o alloc_cpp.o barrier.o beginend.o clone.o eh_cpp.o local.o query.o retry.o rwlock.o useraction.o util.o sjlj.o tls.o method-serial.o method-gl.o method-ml.o futex.o
libtool: link: /fork.openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/aarch64-openwrt-linux-gnu/bin/ranlib --plugin /fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0-final/./gcc/liblto_plugin.so .libs/libitm.a
libtool: link: ( cd ".libs" && rm -f "libitm.la" && ln -s "../libitm.la" "libitm.la" )
true  DO=all multi-do # make
make[7]: Leaving directory '/fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0-final/aarch64-openwrt-linux-gnu/libitm'
make[6]: Leaving directory '/fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0-final/aarch64-openwrt-linux-gnu/libitm'
make[5]: Leaving directory '/fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0-final/aarch64-openwrt-linux-gnu/libitm'
Checking multilib configuration for libatomic...
make[5]: Entering directory '/fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0-final/aarch64-openwrt-linux-gnu/libatomic'
make  all-recursive
make[6]: Entering directory '/fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0-final/aarch64-openwrt-linux-gnu/libatomic'
Making all in testsuite
make[7]: Entering directory '/fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0-final/aarch64-openwrt-linux-gnu/libatomic/testsuite'
make[7]: Nothing to be done for 'all'.
make[7]: Leaving directory '/fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0-final/aarch64-openwrt-linux-gnu/libatomic/testsuite'
make[7]: Entering directory '/fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0-final/aarch64-openwrt-linux-gnu/libatomic'
/bin/bash ./libtool  --tag=CC   --mode=compile /fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0-final/./gcc/xgcc -B/fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0-final/./gcc/ -B/fork.openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/aarch64-openwrt-linux-gnu/bin/ -B/fork.openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/aarch64-openwrt-linux-gnu/lib/ -isystem /fork.openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/aarch64-openwrt-linux-gnu/include -isystem /fork.openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/aarch64-openwrt-linux-gnu/sys-include    -DHAVE_CONFIG_H -I/fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0/libatomic/config/aarch64 -I/fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0/libatomic/config/linux/aarch64 -I/fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0/libatomic/config/posix -I/fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0/libatomic -I.    -mno-outline-atomics -Wall -Werror   -pthread -O2 -pipe -mcpu=cortex-a53 -fno-caller-saves -fno-plt -fhonour-copts -Wformat -Werror=format-security -g -Os -MT load_1_1_.lo -MD -MP -MF .deps/load_1_1_.lo.Ppo -DN=1 -DIFUNC_ALT=1 -march=armv8-a+lse -c -o load_1_1_.lo /fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0/libatomic/load_n.c
libtool: compile:  /fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0-final/./gcc/xgcc -B/fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0-final/./gcc/ -B/fork.openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/aarch64-openwrt-linux-gnu/bin/ -B/fork.openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/aarch64-openwrt-linux-gnu/lib/ -isystem /fork.openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/aarch64-openwrt-linux-gnu/include -isystem /fork.openwrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/aarch64-openwrt-linux-gnu/sys-include -DHAVE_CONFIG_H -I/fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0/libatomic/config/aarch64 -I/fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0/libatomic/config/linux/aarch64 -I/fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0/libatomic/config/posix -I/fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0/libatomic -I. -mno-outline-atomics -Wall -Werror -pthread -O2 -pipe -mcpu=cortex-a53 -fno-caller-saves -fno-plt -fhonour-copts -Wformat -Werror=format-security -g -Os -MT load_1_1_.lo -MD -MP -MF .deps/load_1_1_.lo.Ppo -DN=1 -DIFUNC_ALT=1 -march=armv8-a+lse -c /fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0/libatomic/load_n.c  -fPIC -DPIC -o .libs/load_1_1_.o
cc1: error: switch '-mcpu=cortex-a53' conflicts with '-march=armv8-a+lse' switch and resulted in options '+lse+nocrc' being added [-Werror]
cc1: all warnings being treated as errors
make[7]: *** [Makefile:899: load_1_1_.lo] Error 1
make[7]: Leaving directory '/fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0-final/aarch64-openwrt-linux-gnu/libatomic'
make[6]: *** [Makefile:650: all-recursive] Error 1
make[6]: Leaving directory '/fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0-final/aarch64-openwrt-linux-gnu/libatomic'
make[5]: *** [Makefile:468: all] Error 2
make[5]: Leaving directory '/fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0-final/aarch64-openwrt-linux-gnu/libatomic'
make[4]: *** [Makefile:16994: all-target-libatomic] Error 2
make[4]: Leaving directory '/fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0-final'
make[3]: *** [Makefile:1063: all] Error 2
make[3]: Leaving directory '/fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0-final'
make[2]: *** [Makefile:92: /fork.openwrt/build_dir/toolchain-aarch64_cortex-a53_gcc-14.2.0_glibc/gcc-14.2.0-final/.built] Error 2
make[2]: Leaving directory '/fork.openwrt/toolchain/gcc/final'
time: toolchain/gcc/final/compile#329.25#26.75#341.78
    ERROR: toolchain/gcc/final failed to build.
make[1]: *** [toolchain/Makefile:93: toolchain/gcc/final/compile] Error 1
make[1]: Leaving directory '/fork.openwrt'
make: *** [/fork.openwrt/include/toplevel.mk:233: toolchain/gcc/final/compile] Error 2
```

Any ideas?

--- a/toolchain/gcc/Config.in
+++ b/toolchain/gcc/Config.in
@@ -2,7 +2,7 @@
 
 choice
 	prompt "GCC compiler Version" if TOOLCHAINOPTS
-	default GCC_USE_VERSION_13
+	default GCC_USE_VERSION_14
 	help
 	  Select the version of gcc you wish to use.
 

--- a/toolchain/gcc/Config.version
+++ b/toolchain/gcc/Config.version
@@ -6,8 +6,8 @@ config GCC_VERSION_12
 	default y if GCC_USE_VERSION_12
 	bool
 
-config GCC_VERSION_14
-	default y if GCC_USE_VERSION_14
+config GCC_VERSION_13
+	default y if GCC_USE_VERSION_13
 	bool
 
 config GCC_VERSION
@@ -15,10 +15,10 @@ config GCC_VERSION
 	default EXTERNAL_GCC_VERSION	if EXTERNAL_TOOLCHAIN && !NATIVE_TOOLCHAIN
 	default "11.3.0"	if GCC_VERSION_11
 	default "12.3.0"	if GCC_VERSION_12
-	default "14.2.0"	if GCC_VERSION_14
-	default "13.3.0"
+	default "13.3.0"	if GCC_VERSION_13
+	default "14.2.0"
 
 config GCC_USE_DEFAULT_VERSION
 	bool
-	default y if !TOOLCHAINOPTS || GCC_USE_VERSION_13
+	default y if !TOOLCHAINOPTS || GCC_USE_VERSION_14
 	imply KERNEL_WERROR

